### PR TITLE
Add `"json"` destination, for JSON modules

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
The import attributes TC39 proposal has been updated so that HTML can use the expected type (unspecified, `json`, `css`) to affect the fetch request. This PR introduces the `"json"` destination, that will be used to fetch `import "..." with { type: "json" }`.

Ref https://github.com/whatwg/html/pull/9486#discussion_r1273195679, https://github.com/whatwg/html/issues/7233

- [x] At least two implementers are interested (and none opposed):
   * Chromium
   * WebKit
- [x] [Tests](https://github.com/web-platform-tests/wpt) are written and can be reviewed and commented upon at:
   * https://github.com/web-platform-tests/wpt/pull/41665
- [x] [Implementation bugs](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) are filed:
   * Chromium: https://bugs.chromium.org/p/chromium/issues/detail?id=1491336
   * Gecko: https://bugzilla.mozilla.org/show_bug.cgi?id=1858078
   * WebKit: https://bugs.webkit.org/show_bug.cgi?id=262946
   * Deno: https://github.com/denoland/deno/issues/20866
   * Node.js: https://github.com/nodejs/node/issues/50116
- [x] [MDN issue](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) is filed: https://github.com/mdn/content/pull/29573

(See [WHATWG Working Mode: Changes](https://whatwg.org/working-mode#changes) for more details.)

***
<a href="https://whatpr.org/whatwg-fetch/1691.html" title="Last updated on Oct 29, 2023, 8:05 PM UTC (5e68603)">Preview</a> | <a href="https://whatpr.org/fetch/1691/9952a11...5e68603.html" title="Last updated on Oct 29, 2023, 8:05 PM UTC (5e68603)">Diff</a>

### CODE REVIEW

File: fetch.bs [hunk 1]
```diff
@@ -1725,6 +1725,7 @@ the empty string,
 "<code>frame</code>",
 "<code>iframe</code>",
 "<code>image</code>",
+"<code>json</code>",
```
Comment (annevk) [inline fetch.bs:1728]: You'll need to update `destination-table` below as well.

**Review by annevk — COMMENTED** [review]
Looks great modulo some nits.

I think we can count Chromium and WebKit as supportive based on statements in https://github.com/whatwg/html/issues/7233, but I'll tag some people here just in case.

cc @Constellation @syg @hiroshige-g

**Comment by annevk** [comment]
@nicolo-ribaudo are you planning to write tests for this and implementation bugs?

**Comment by hiroshige-g** [comment]
cc/ @domenic

**Comment by nicolo-ribaudo** [comment]
Tests have been merged!

**Pull request merged**